salt/modules/virt.py

```diff
@@ -599,18 +601,29 @@ def _gen_xml(
                 context["boot"]["kernel"] = "/usr/lib/grub2/x86_64-xen/grub.xen"
                 context["boot_dev"] = []
 
-    if "serial_type" in kwargs:
-        context["serial_type"] = kwargs["serial_type"]
-    if "serial_type" in context and context["serial_type"] == "tcp":
-        if "telnet_port" in kwargs:
-            context["telnet_port"] = kwargs["telnet_port"]
-        else:
-            context["telnet_port"] = 23023  # FIXME: use random unused port
-    if "serial_type" in context:
-        if "console" in kwargs:
-            context["console"] = kwargs["console"]
-        else:
-            context["console"] = True
```

Comment:
Here you cannot get rid of the old parameters so easily since we need to flag them as deprecated before removing them to avoid being to harsh to people using them. You need to do something like this:

```python
default_serial_type = kwargs.get("serial_type")
if default_serial_type:
    salt.utils.versions.warn_until(
        "Aluminium",
        "'serial_type' parameter has been deprecated, use the 'serials' and 'consoles' parameters instead. "
        "It will be removed in {version}.",
    )

default_port = kwargs.get("telnet_port")
if telnet_port:
    salt.utils.versions.warn_until(
        "Aluminium",
        "'telnet_port' parameter has been deprecated, use the 'serials' and 'consoles' parameters instead. "
        "It will be removed in {version}.",
    )
else:
    default_port = 23023

# TODO do something similar for 'console' parameter
```